samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -317,15 +317,55 @@ public Map<TaskName, Map<SystemStreamPartition, Startpoint>> fanOut(Map<TaskName
    * @return fanned out Startpoints
    */
   public Map<SystemStreamPartition, Startpoint> getFanOutForTask(TaskName taskName) throws IOException {
+    return getStartpointFanOutPerTask(taskName)
```

Comment:
@alnzng  Do we think it will be useful to emit a metric here when we are performing the CRUD operations on these fanout tasks(Like deleting a startpoint at a task level). 

This will be useful because typically we rely upon logs of container runs to determine if the startpoint resolution/deletion had worked as expected. These logs will get purged after TTL expiration.

 If we agree to publish metrics, then it will help better monitor for samza jobs in production

Reply:
In general, metrics make more sense than logging if the monitoring behaviors/operations are producing time-series data, which means the data are kept updated frequently in the app's lifecycle. Either metrics and logs can have their own TTL, which depends on the SLA for their backend storage services. Also, one con for metrics is that they can't include too much information for troubleshooting because of the high-cardinality challenge in the metrics storage.

For this particular use case "startpoints", it was intended for once in life and all its related operations are usually only invoked very few times in the very early phase of the app's lifecycle. I would prefer to keep using logs than metrics since we can add as much as information we want to help troubleshoot.